sinks/splunk/splunk.go

```diff
@@ -189,6 +189,8 @@ func (sss *splunkSpanSink) submitter(sync chan struct{}, signalReady sync.Once,
 
 		hecReq, err := sss.hec.newRequest()
 
+		encodeErrors := make(chan error)
```

Comment:
One thing I'm not sure about is what will happen to the `encodeErrors` chans when a `batchTimeout` happens - will this leave a chan with no reader around & will the goroutine that tries to send the write error just leak?

Cancelation without contexts is hard, you all

Reply:
yeah it looks like there's a goroutine leak if a timeout happens, because the encoder will get an error (trying to write to a closed pipe) and it'll try to send to a channel that never exists. maybe give the channel a buffer of 1? the loop always receives from the channel before spawning another goroutine to send to it